### PR TITLE
[device_info_plus] Detect iOS simulator model identifier

### PR DESCRIPTION
**Pull request by shaynekellyii**
## Description

Currently, `iosInfo.utsname.machine` returns the underlying architecture of the iOS device when the device is a simulator. This updates the behavior to return the simulator's model identifier instead (e.g. `iPhone 15,2` for an iPhone 14 Pro).

## Related Issues

https://github.com/fluttercommunity/plus_plugins/issues/273

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by vbuberen** [comment]
Thanks for your contribution. Looks good to me, but would like to also hear opinion of other maintainers first.

**Comment by vbuberen** [comment]
Also, please fix issues detected by checks, like code formatting.

**Comment by miquelbeltran** [comment]
LMTG, needs the formatting fixed, and we can merge it.

Would it be possible to have an integration test for this as well? The setup is already done. The GitHub Action uses the iPhone 8 model on the simulator.

**Comment by shaynekellyii** [comment]
Formatted the objc code, and added a test to check that all iOS utsname fields are non-null

File: packages/device_info_plus/device_info_plus/example/integration_test/device_info_plus_test.dart [hunk 1]
```diff
@@ -67,6 +67,17 @@ void main() {
...
+  testWidgets('Can get non-null iOS utsname fields',
+      (WidgetTester tester) async {
+    if (Platform.isIOS) {
+      expect(iosInfo.utsname.machine, 'iPhone10,4');
+      expect(iosInfo.utsname.nodename, isNotNull);
+      expect(iosInfo.utsname.release, isNotNull);
+      expect(iosInfo.utsname.sysname, isNotNull);
+      expect(iosInfo.utsname.version, isNotNull);
+    }
+  });
```
Comment (miquelbeltran) [inline packages/device_info_plus/device_info_plus/example/integration_test/device_info_plus_test.dart:79]: Can you make the following changes:

- remove the if condition
- add `skip: !Platform.isIOS` to the test

Comment (shaynekellyii) [reply packages/device_info_plus/device_info_plus/example/integration_test/device_info_plus_test.dart]: Updated 👍

**Comment by shaynekellyii** [comment]
Hmm, not sure why `package_info_plus_windows` is failing to bootstrap since I haven't made any changes there.

**Comment by miquelbeltran** [comment]
Thank you!

**Review by miquelbeltran — APPROVED** [review]

**Pull request merged**